hcl/hcl.go

```diff
@@ -25,8 +25,8 @@ type Blocks interface {
 	*Host | *Product | *Agent
 }
 
-// NOTE(dcohen) this is currently a separate config block, as opposed to a parent block of the others
 type Agent struct {
+	// NOTE(dcohen) this is currently a separate config block, as opposed to a parent block of the others
```

Comment:
The comment directly above an exported object is supposed to be a doc comment, which starts with the name of the object. So, since this one's a note, I pulled it down inside the struct.